pulsar-impl/src/main/java/com/datastax/oss/sink/pulsar/PulsarSchema.java

```diff
@@ -74,6 +77,11 @@ public static PulsarSchema of(String path, Object value, LocalSchemaRegistry reg
     if (value instanceof Short) {
       return INT16;
     }
+    // Mark Lists/Maps as STRUCT to leverage StructToJsonNodeCodecAdapter for Avro schemas.
+    // In Json schemas, List/Map will be of type GenericRecord.
+    if (value instanceof List || value instanceof Map) {
+      return STRUCT;
```

Comment:
can we have LIST and STRUCT ?
it will be somehow clearer

a LIST is quite different from a STRUCT

Reply:
You are right about the clarity. The issue with List though is our RecordMapper logic will select a codec based on this type and by using STRUCT here (and wrapping list/array with struct when we access PulsarStruct.get("fieldName") we are able to take advantage of the Struct based codecs. This is parallel in accept to how Json schema handles array in maps - there are both wrapped in struct objects implicitly. 

So the mapper codec [selection logic](https://github.com/datastax/messaging-connectors-commons/blob/cf4c470b2c180bc35c1a6053c4e0b27bb6bf6c4b/common/src/main/java/com/datastax/oss/common/sink/RecordMapper.java#L298) is base on the [schema](https://github.com/datastax/messaging-connectors-commons/blob/cf4c470b2c180bc35c1a6053c4e0b27bb6bf6c4b/common/src/main/java/com/datastax/oss/common/sink/RecordMapper.java#L193), and the codec requires a PulsarStruct object at runtime so both pieces are required to complete the puzzle.

Reply:
Please check the latest patch. I resembled the json schema behavior closer:
* Now the pulsar schema struct class properly generates generic records to wrap avro array and lists. Generic records are already handled by the code in Pulsar schema and there is no need to hack the STRUCT type anymore.
* GenericRecordWrapper is now updated to populated the fields of the Pulsar structs and enables nested maps/arrays as well. I updated the e2e tests to now check for UDTs mapped from Maps (in addition to Pojo in the old patch)
* There is 1 extra conversion required (check references of shouldWrapAvroType) - this is required for max code reusability & correctness. The Pulsar schema updates & Pulsar Struct get are separate code paths and both required to wrapped version to work correctly.